assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -704,13 +566,14 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
     60,
     (W - SLOT_WIDTH * numStops) / (numStops - 1)
   );
-  const { 0: beginning, [slots.length - 1]: end, ...middle } = slots;
+  const [beginning, middle, end] = [slots[0], slots.slice(1, -1), slots.at(-1)];
   const hasEmphasis = effect !== "station_closure";
+  const calculated_emphasis_height = hasEmphasis ? EMPHASIS_HEIGHT + EMPHASIS_PADDING_TOP : 0;
   const labelTextClass = slots.length > 12 ? "small" : "large";
 
   let x = 0;
   const middleSlots = Object.values(middle).map((s, i) => {
-    x = (spaceBetween + SLOT_WIDTH) * (i + 1) + L;
+    x = (spaceBetween + SLOT_WIDTH) * (i + 1);
```

Comment:
Since you use `i + 1` throughout this function body to offset the removal of `beginning` from the array, consider assigning that to a variable, maybe like:
```ts
const middleSlots = middle.map((s, i) => {
  // Add 1 to the index to counteract the offset caused by removing `beginning` from the original `slots` array.
  const slotIndex = i + 1;
  x = (spaceBetween + SLOT_WIDTH) * slotIndex;
  ... // replace all the other `x + 1`'s with `slotIndex`
})
```